### ISSUE TITLE
Bug/Parser: Log files not parsed, no error message given

### ISSUE THREAD

**Issue by rbqbjyur** [crowdsecurity/crowdsec#1097]
**Technical Information (please complete the following information):**
 - OS: CentOS Stream 8
 - CrowdSec v1.2.1
 

**Describe the bug**
Just installed CrowdSec v1.2.1 this morning on CentOS Stream 8, CrowdSec and Bouncer services running.
Collections installed, and up to date.

However while testing if it worked, I noted that it wouldn't detect any failed SSH attempts, and so wouldn't be able to block.
This appears to be due to `crowdsecurity/sshd-logs` and/or `/var/log/secure` being entirely unparsed, with no apparent error messages given.

Have just read up on CrowdSec today, so apologies for my unfamiliarity/dump of information.

**To Reproduce**
Installed CrowdSec
```
curl -s https://packagecloud.io/install/repositories/crowdsec/crowdsec/script.rpm.sh | sudo bash
sudo dnf install crowdsec
sudo systemctl enable --now crowdsec
```

Ensured was running
`sudo systemctl status crowdsec`
```
● crowdsec.service - Crowdsec agent
   Loaded: loaded (/usr/lib/systemd/system/crowdsec.service; enabled; vendor preset: enabled)
   Active: active (running) since Tue 2021-12-14 13:28:13 GMT; 29s ago
  Process: 2492560 ExecReload=/bin/kill -HUP $MAINPID (code=exited, status=0/SUCCESS)
  Process: 2539360 ExecStartPre=/usr//bin/crowdsec -c /etc/crowdsec/config.yaml -t (code=exited, status=0/SUCCESS)
 Main PID: 2539375 (crowdsec)
    Tasks: 18 (limit: 48634)
   Memory: 29.1M
   CGroup: /system.slice/crowdsec.service
           ├─2539375 /usr/bin/crowdsec -c /etc/crowdsec/config.yaml
           ├─2539409 journalctl --follow -n 0 _SYSTEMD_UNIT=mysql.service
           └─2539410 journalctl --follow -n 0 _SYSTEMD_UNIT=apache2.service

Dec 14 13:28:12 colossus systemd[1]: Starting Crowdsec agent...
Dec 14 13:28:13 colossus systemd[1]: Started Crowdsec agent.
```

Installed Bouncer
```
sudo dnf install crowdsec-firewall-bouncer-iptables
sudo systemctl enable --now crowdsec-firewall-bouncer
```

Ensured was running
`sudo systemctl status crowdsec-firewall-bouncer`
```
● crowdsec-firewall-bouncer.service - The firewall bouncer for CrowdSec
   Loaded: loaded (/usr/lib/systemd/system/crowdsec-firewall-bouncer.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2021-12-14 13:28:23 GMT; 55s ago
  Process: 2539683 ExecStartPost=/bin/sleep 0.1 (code=exited, status=0/SUCCESS)
  Process: 2539606 ExecStartPre=/usr/bin/crowdsec-firewall-bouncer -c /etc/crowdsec/bouncers/crowdsec-firewall-bouncer.yaml -t (code=exited, status=0/SUCCESS)
 Main PID: 2539613 (crowdsec-firewa)
    Tasks: 13 (limit: 48634)
   Memory: 9.7M
   CGroup: /system.slice/crowdsec-firewall-bouncer.service
           └─2539613 /usr/bin/crowdsec-firewall-bouncer -c /etc/crowdsec/bouncers/crowdsec-firewall-bouncer.yaml

Dec 14 13:28:21 colossus systemd[1]: Starting The firewall bouncer for CrowdSec...
Dec 14 13:28:21 colossus crowdsec-firewall-bouncer[2539606]: time="2021-12-14T13:28:21Z" level=info msg="crowdsec-firewall-bouncer v0.0.16-el8-rpm-ea64223d4e41c28b5d6324e82ab07625874ce507"
Dec 14 13:28:21 colossus crowdsec-firewall-bouncer[2539613]: time="2021-12-14T13:28:21Z" level=info msg="crowdsec-firewall-bouncer v0.0.16-el8-rpm-ea64223d4e41c28b5d6324e82ab07625874ce507"
Dec 14 13:28:23 colossus systemd[1]: Started The firewall bouncer for CrowdSec.
```

Updated hub
```
$ sudo cscli hub update
INFO[14-12-2021 01:30:53 PM] Wrote new 190449 bytes index to /etc/crowdsec/hub/.index.json
$ sudo cscli hub upgrade
INFO[14-12-2021 01:30:57 PM] Upgrading collections
INFO[14-12-2021 01:30:57 PM] crowdsecurity/linux : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/base-http-scenarios : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/nginx : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/apache2 : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/mysql : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/sshd : up-to-date
INFO[14-12-2021 01:30:57 PM] All collections are already up-to-date
INFO[14-12-2021 01:30:57 PM] Upgrading parsers
INFO[14-12-2021 01:30:57 PM] crowdsecurity/syslog-logs : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/whitelists : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/geoip-enrich : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/mysql-logs : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/sshd-logs : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/dateparse-enrich : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/nginx-logs : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/apache2-logs : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-logs : up-to-date
INFO[14-12-2021 01:30:57 PM] All parsers are already up-to-date
INFO[14-12-2021 01:30:57 PM] Upgrading scenarios
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-crawl-non_statics : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-bad-user-agent : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-open-proxy : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-sqli-probing : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/ssh-slow-bf : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-probing : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-xss-probing : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/ssh-bf : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-backdoors-attempts : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-generic-bf : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/mysql-bf : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-path-traversal-probing : up-to-date
INFO[14-12-2021 01:30:57 PM] crowdsecurity/http-sensitive-files : up-to-date
INFO[14-12-2021 01:30:57 PM] ltsich/http-w00tw00t : up-to-date
INFO[14-12-2021 01:30:57 PM] All scenarios are already up-to-date
INFO[14-12-2021 01:30:57 PM] Upgrading postoverflows
INFO[14-12-2021 01:30:57 PM] No postoverflows installed, nothing to upgrade
```

From another computer, attempted to SSH, intentionally with incorrect username or password multiple times.
Attempts appear in `/var/log/secure`
```
Dec 14 12:50:46 hostname sshd[2475575]: error: PAM: Authentication failure for user from 192.168.1.4
Dec 14 12:50:46 hostname sshd[2475575]: Postponed keyboard-interactive for user from 192.168.1.4 port 55264 ssh2 [preauth]
Dec 14 12:50:47 hostname sshd[2475575]: Connection closed by authenticating user user 192.168.1.4 port 55264 [preauth]
```

but not in `/var/log/crowdsec.log`.
```
$ tail /var/log/crowdsec.log
time="14-12-2021 13:28:13" level=info msg="loading acquisition file : /etc/crowdsec/acquis.yaml"
time="14-12-2021 13:28:13" level=info msg="Adding file /var/log/nginx/access.log to datasources" type=file
time="14-12-2021 13:28:13" level=info msg="Adding file /var/log/nginx/error.log to datasources" type=file
time="14-12-2021 13:28:13" level=info msg="Adding file /var/log/secure to datasources" type=file
time="14-12-2021 13:28:13" level=info msg="Adding file /var/log/messages to datasources" type=file
time="14-12-2021 13:28:13" level=warning msg="Starting processing data"
time="14-12-2021 13:28:13" level=info msg="Running journalctl command: /usr/bin/journalctl [journalctl --follow -n 0 _SYSTEMD_UNIT=mysql.service]" src="journalctl-_SYSTEMD_UNIT=mysql.service" type=journalctl
time="14-12-2021 13:28:13" level=info msg="Running journalctl command: /usr/bin/journalctl [journalctl --follow -n 0 _SYSTEMD_UNIT=apache2.service]" src="journalctl-_SYSTEMD_UNIT=apache2.service" type=journalctl
time="14-12-2021 13:28:14" level=info msg="capi metrics: metrics sent successfully"
time="14-12-2021 13:28:14" level=info msg="start crowdsec api send metrics (interval: 30m)"
```

Also ssh-logs appear to be unparsed from `sudo cscli metrics` (full output at end)
```
INFO[14-12-2021 01:13:56 PM] Acquisition Metrics:                         
+-----------------------------------------------------+------------+--------------+----------------+------------------------+
|                       SOURCE                        | LINES READ | LINES PARSED | LINES UNPARSED | LINES POURED TO BUCKET |
+-----------------------------------------------------+------------+--------------+----------------+------------------------+
| file:/var/log/secure                                |        176 | -            |            176 | -                      |
+-----------------------------------------------------+------------+--------------+----------------+------------------------+
INFO[14-12-2021 01:13:56 PM] Parser Metrics:                              
+----------------------------------+------+--------+----------+
|             PARSERS              | HITS | PARSED | UNPARSED |
+----------------------------------+------+--------+----------+
| child-crowdsecurity/sshd-logs    |   84 | -      |       84 |
| crowdsecurity/sshd-logs          |   14 | -      |       14 |
+----------------------------------+------+--------+----------+
```

I imagine this is the key part as these files would need to be parsed to be detected.
However I am unsure how to diagnose what causes files to be labelled "UNPARSED", if there is any command or log I can check, please let me know and I'll try to figure it out.

**Expected behaviour**
Either Log files are parsed or way to find cause for not parsing

**Logs**
```
$ sudo cscli metrics
INFO[14-12-2021 01:13:56 PM] Buckets Metrics:                             
+--------------------------------------+---------------+-----------+--------------+--------+---------+
|                BUCKET                | CURRENT COUNT | OVERFLOWS | INSTANCIATED | POURED | EXPIRED |
+--------------------------------------+---------------+-----------+--------------+--------+---------+
| crowdsecurity/http-crawl-non_statics | -             | -         |            1 |      1 |       1 |
| crowdsecurity/http-probing           | -             | -         |            1 |      1 |       1 |
+--------------------------------------+---------------+-----------+--------------+--------+---------+
INFO[14-12-2021 01:13:56 PM] Acquisition Metrics:                         
+-----------------------------------------------------+------------+--------------+----------------+------------------------+
|                       SOURCE                        | LINES READ | LINES PARSED | LINES UNPARSED | LINES POURED TO BUCKET |
+-----------------------------------------------------+------------+--------------+----------------+------------------------+
| file:/var/log/messages                              |        895 | -            |            895 | -                      |
| file:/var/log/nginx/access.log                      |         39 |           38 |              1 |                      2 |
| file:/var/log/nginx/error.log                       |          3 | -            |              3 | -                      |
| file:/var/log/secure                                |        176 | -            |            176 | -                      |
| journalctl:journalctl-_SYSTEMD_UNIT=apache2.service |          2 | -            |              2 | -                      |
| journalctl:journalctl-_SYSTEMD_UNIT=mysql.service   |          2 | -            |              2 | -                      |
+-----------------------------------------------------+------------+--------------+----------------+------------------------+
INFO[14-12-2021 01:13:56 PM] Parser Metrics:                              
+----------------------------------+------+--------+----------+
|             PARSERS              | HITS | PARSED | UNPARSED |
+----------------------------------+------+--------+----------+
| child-crowdsecurity/apache2-logs |    4 | -      |        4 |
| child-crowdsecurity/http-logs    |  114 |     99 |       15 |
| child-crowdsecurity/nginx-logs   |   46 |     38 |        8 |
| child-crowdsecurity/sshd-logs    |   84 | -      |       84 |
| crowdsecurity/apache2-logs       |    2 | -      |        2 |
| crowdsecurity/dateparse-enrich   |   38 |     38 | -        |
| crowdsecurity/geoip-enrich       |   38 |     38 | -        |
| crowdsecurity/http-logs          |   38 |     36 |        2 |
| crowdsecurity/mysql-logs         |    2 | -      |        2 |
| crowdsecurity/nginx-logs         |   42 |     38 |        4 |
| crowdsecurity/non-syslog         |   46 |     46 | -        |
| crowdsecurity/sshd-logs          |   14 | -      |       14 |
| crowdsecurity/syslog-logs        | 1071 |   1071 | -        |
| crowdsecurity/whitelists         |   38 |     38 | -        |
+----------------------------------+------+--------+----------+
INFO[14-12-2021 01:13:56 PM] Local Api Metrics:                           
+----------------------+--------+------+
|        ROUTE         | METHOD | HITS |
+----------------------+--------+------+
| /v1/alerts           | GET    |    6 |
| /v1/decisions/stream | GET    |   85 |
| /v1/watchers/login   | POST   |   10 |
+----------------------+--------+------+
INFO[14-12-2021 01:13:56 PM] Local Api Machines Metrics:                  
+--------------------------------------------------+------------+--------+------+
|                     MACHINE                      |   ROUTE    | METHOD | HITS |
+--------------------------------------------------+------------+--------+------+
| 3002ae572102481f80e4a590985e6791zNzWcJMEeHN8yYEd | /v1/alerts | GET    |    6 |
+--------------------------------------------------+------------+--------+------+
INFO[14-12-2021 01:13:56 PM] Local Api Bouncers Metrics:                  
+----------------------------+----------------------+--------+------+
|          BOUNCER           |        ROUTE         | METHOD | HITS |
+----------------------------+----------------------+--------+------+
| FirewallBouncer-1639486791 | /v1/decisions/stream | GET    |   85 |
+----------------------------+----------------------+--------+------+
```

```
$ sudo cscli hub list
INFO[14-12-2021 02:00:05 PM] Loaded 26 collecs, 32 parsers, 45 scenarios, 3 post-overflow parsers 
INFO[14-12-2021 02:00:05 PM] PARSERS:                                     
-------------------------------------------------------------------------------------------------------------
 NAME                            📦 STATUS   VERSION  LOCAL PATH                                             
-------------------------------------------------------------------------------------------------------------
 crowdsecurity/syslog-logs       ✔️  enabled  0.7     /etc/crowdsec/parsers/s00-raw/syslog-logs.yaml         
 crowdsecurity/nginx-logs        ✔️  enabled  0.9      /etc/crowdsec/parsers/s01-parse/nginx-logs.yaml        
 crowdsecurity/mysql-logs        ✔️  enabled  0.3     /etc/crowdsec/parsers/s01-parse/mysql-logs.yaml        
 crowdsecurity/sshd-logs         ✔️  enabled  1.3     /etc/crowdsec/parsers/s01-parse/sshd-logs.yaml         
 crowdsecurity/dateparse-enrich  ✔️  enabled  0.1      /etc/crowdsec/parsers/s02-enrich/dateparse-enrich.yaml 
 crowdsecurity/http-logs         ✔️  enabled  0.6     /etc/crowdsec/parsers/s02-enrich/http-logs.yaml        
 crowdsecurity/geoip-enrich      ✔️  enabled  0.2      /etc/crowdsec/parsers/s02-enrich/geoip-enrich.yaml     
 crowdsecurity/whitelists        ✔️  enabled  0.2      /etc/crowdsec/parsers/s02-enrich/whitelists.yaml      
 crowdsecurity/apache2-logs      ✔️  enabled  0.9     /etc/crowdsec/parsers/s01-parse/apache2-logs.yaml      
-------------------------------------------------------------------------------------------------------------
INFO[14-12-2021 02:00:05 PM] SCENARIOS:                                   
--------------------------------------------------------------------------------------------------------------------------
 NAME                                       📦 STATUS   VERSION  LOCAL PATH                                               
--------------------------------------------------------------------------------------------------------------------------
 crowdsecurity/http-xss-probing             ✔️  enabled  0.2      /etc/crowdsec/scenarios/http-xss-probing.yaml            
 crowdsecurity/mysql-bf                     ✔️  enabled  0.1     /etc/crowdsec/scenarios/mysql-bf.yaml                    
 crowdsecurity/http-bad-user-agent          ✔️  enabled  0.4     /etc/crowdsec/scenarios/http-bad-user-agent.yaml         
 crowdsecurity/http-sensitive-files         ✔️  enabled  0.2     /etc/crowdsec/scenarios/http-sensitive-files.yaml        
 ltsich/http-w00tw00t                       ✔️  enabled  0.1     /etc/crowdsec/scenarios/http-w00tw00t.yaml               
 crowdsecurity/http-crawl-non_statics       ✔️  enabled  0.2      /etc/crowdsec/scenarios/http-crawl-non_statics.yaml     
 crowdsecurity/http-path-traversal-probing  ✔️  enabled  0.2     /etc/crowdsec/scenarios/http-path-traversal-probing.yaml 
 crowdsecurity/ssh-slow-bf                  ✔️  enabled 0.2      /etc/crowdsec/scenarios/ssh-slow-bf.yaml                 
 crowdsecurity/http-backdoors-attempts      ✔️  enabled  0.2      /etc/crowdsec/scenarios/http-backdoors-attempts.yaml     
 crowdsecurity/http-generic-bf              ✔️  enabled  0.1     /etc/crowdsec/scenarios/http-generic-bf.yaml             
 crowdsecurity/http-sqli-probing            ✔️  enabled 0.2      /etc/crowdsec/scenarios/http-sqli-probing.yaml           
 crowdsecurity/http-open-proxy              ✔️  enabled 0.2      /etc/crowdsec/scenarios/http-open-proxy.yaml             
 crowdsecurity/http-probing                 ✔️  enabled  0.2     /etc/crowdsec/scenarios/http-probing.yaml                
 crowdsecurity/ssh-bf                       ✔️  enabled  0.1     /etc/crowdsec/scenarios/ssh-bf.yaml                      
--------------------------------------------------------------------------------------------------------------------------
INFO[14-12-2021 02:00:05 PM] COLLECTIONS:                                 
------------------------------------------------------------------------------------------------------------
 NAME                               📦 STATUS   VERSION  LOCAL PATH                                         
------------------------------------------------------------------------------------------------------------
 crowdsecurity/base-http-scenarios  ✔️  enable  0.5      /etc/crowdsec/collections/base-http-scenarios.yaml 
 crowdsecurity/linux                ✔️  enabled  0.2      /etc/crowdsec/collections/linux.yaml               
 crowdsecurity/mysql                ✔️  enabled  0.1     /etc/crowdsec/collections/mysql.yaml               
 crowdsecurity/sshd                 ✔️  enabled  0.2     /etc/crowdsec/collections/sshd.yaml                
 crowdsecurity/nginx                ✔️  enabled  0.1      /etc/crowdsec/collections/nginx.yaml               
 crowdsecurity/apache2              ✔️  enabled  0.1      /etc/crowdsec/collections/apache2.yaml             
------------------------------------------------------------------------------------------------------------
INFO[14-12-2021 02:00:05 PM] POSTOVERFLOWS:                               
--------------------------------------
 NAME  📦 STATUS  VERSION  LOCAL PATH 
--------------------------------------
--------------------------------------
```

**Comment by buixor** [comment]
Hello @rbqbjyur 

It seems that the logs produced by your centos stream 8 sshd service aren't supported : https://hub.crowdsec.net/author/crowdsecurity/configurations/sshd-logs

Do you know if the ssh logs look the same in stream 8 and stream 9 ? (I'm booting some EC2 to reproduce and fix parser).

**Comment by rbqbjyur** [comment]
Hi, apologies for delay. 

I'm not aware of any sshd log changes between CentOS Stream 8 and 9. But will look into and see if they come out any different.

I wouldn't have though it is unique to CentOS Stream, could the different format also be due to different ssh authentication methods (I require both Key + TOTP, no password accepted)? As you said, looks like it is formatted differently to what the parser expects.

**Comment by rbqbjyur** [comment]
No difference in message (in CentOS Stream 9).

I also tried the same sshd config on a Ubuntu 21.10 server, I get the same format errors in the equivalent log file `/var/log/auth.log`
```
Dec 21 20:02:19 enigma sshd[128449]: error: PAM: Authentication failure for outis from 192.168.1.4
Dec 21 20:02:19 enigma sshd[128449]: Postponed keyboard-interactive for outis from 192.168.1.4 port 55584 ssh2 [preauth]
Dec 21 20:02:21 enigma sshd[128449]: Connection closed by authenticating user outis 192.168.1.4 port 55584 [preauth]
Dec 21 20:02:44 enigma sshd[129113]: User root from 192.168.1.4 not allowed because not listed in AllowUsers
Dec 21 20:02:44 enigma sshd[129113]: Connection closed by invalid user root 192.168.1.4 port 55586 [preauth]
```

So nothing to do with CentOS Stream 8/9

This is also with the recent sshd-logs 1.5 

Similarly on Ubuntu, nothing shown in `/var/log/crowdsec.log`
```
outis@enigma:~$ tail /var/log/crowdsec.log
time="21-12-2021 20:01:08" level=info msg="Adding leaky bucket" cfg=dawn-wood file=/etc/crowdsec/scenarios/ssh-bf.yaml name=crowdsecurity/ssh-bf_user-enum
time="21-12-2021 20:01:08" level=info msg="Adding leaky bucket" cfg=shy-fire file=/etc/crowdsec/scenarios/ssh-slow-bf.yaml name=crowdsecurity/ssh-slow-bf
time="21-12-2021 20:01:08" level=info msg="Adding leaky bucket" cfg=weathered-violet file=/etc/crowdsec/scenarios/ssh-slow-bf.yaml name=crowdsecurity/ssh-slow-bf_user-enum
time="21-12-2021 20:01:08" level=warning msg="Loaded 4 scenarios"
time="21-12-2021 20:01:08" level=info msg="loading acquisition file : /etc/crowdsec/acquis.yaml"
time="21-12-2021 20:01:08" level=info msg="Adding file /var/log/auth.log to datasources" type=file
time="21-12-2021 20:01:08" level=info msg="Adding file /var/log/syslog to datasources" type=file
time="21-12-2021 20:01:08" level=info msg="Adding file /var/log/kern.log to datasources" type=file
time="21-12-2021 20:01:08" level=info msg="Reload is finished"
time="21-12-2021 20:01:08" level=warning msg="Starting processing data"
```

```
outis@enigma:~$ sudo cscli metrics
INFO[21-12-2021 08:11:59 PM] Parser Metrics:                              
+-------------------------------+------+--------+----------+
|            PARSERS            | HITS | PARSED | UNPARSED |
+-------------------------------+------+--------+----------+
| child-crowdsecurity/sshd-logs |   65 | -      |       65 |
| crowdsecurity/sshd-logs       |   10 | -      |       10 |
+-------------------------------+------+--------+----------+
```

As different distributions are providing the same error, I imagine it's to do with the SSH config on the server (likely the authentication method).

If you need me to try anything else out let me know, will do.

**Comment by buixor** [comment]
Hello,

Thanks for the report ! It's most likely because your ssh config disallows password auth and the parser didn't catch the :

```
Dec 21 20:02:21 enigma sshd[128449]: Connection closed by authenticating user outis 192.168.1.4 port 55584 [preauth]
```

It's being fixed by https://github.com/crowdsecurity/hub/pull/324

As soon as it's merged (if you're in the latest version) and you run `cscli hub update && cscli hub upgrade`, it should be solved !
